neuralmonkey/decoders/encoder_projection.py

```diff
@@ -65,9 +65,10 @@ def func(train_mode: tf.Tensor,
                 "You must supply rnn_size for this type of encoder projection")
 
         en_concat = concat_encoder_projection(train_mode, None, encoders)
-        en_concat = dropout(en_concat, dropout_keep_prob, train_mode)
 
-        return tf.layers.dense(en_concat, rnn_size, name="encoders_projection")
+        return dropout(
```

Comment:
Myslel jsme, že máme koncenci, že vracíme všechno bez dropoutu a dropoutujeme vstupy a ne výstupy. @jindrahelcl?

Reply:
V transformerech dropoutujeme v return statementu. Tenhle koncept se mi libi vic, tak jsem se ho drzel.

Reply:
Hm, to asi děláme protože je to opsaný s Tensor2Tensor.

Reply:
Ale v těch rekuretních modelech IMHO skoro všude dropoutujeme vstupy.

Reply:
úzus byl dělat dropout na začátku. Vždycky jsem o tom přemýšlel tak, že výstup má bejt plnohodnotnej výstup a jestli si ho chci vydropoutovat na vstupu další komponenty, má za to bejt zodpovědná ta další.

Reply:
a transformer je opsanej z t2t a tam dělaj ty jejich divný preprocesy a postprocesy, tak jsem to nechtěl zbytečně komplikovat tim, že bych to refaktoroval na náš úzus, abychom mohli ten kód porovnávat.

Reply:
Droupout by mela byt vlastnost vystupu, protoze to souvisi s trenovanim te dane vystupni vrstvy vrstvy.

Btw, zajimalo by me, jak by probihalo trenovani vrstvy, ktera leze jako vstup do dvou metod; jedna si ji dropoutne a druha ne.